pulsar-metadata/src/main/java/org/apache/pulsar/metadata/coordination/impl/ResourceLockImpl.java

```diff
@@ -218,18 +205,22 @@ synchronized void lockWasInvalidated() {
         if (revalidateAfterReconnection) {
             revalidateAfterReconnection = false;
             log.warn("Revalidate lock at {} after reconnection", path);
-            return revalidate();
+            return revalidate(value);
         } else {
             return CompletableFuture.completedFuture(null);
         }
     }
 
-    synchronized CompletableFuture<Void> revalidate() {
+    synchronized CompletableFuture<Void> revalidate(T newValue) {
         return store.get(path)
                 .thenCompose(optGetResult -> {
                     if (!optGetResult.isPresent()) {
                         // The lock just disappeared, try to acquire it again
-                        return acquireWithNoRevalidation()
+                        synchronized (ResourceLockImpl.this) {
```

Comment:
May it be useful to have a synchonized setVersion() method to reduce code duplication and it will make code more readable?

Reply:
Makes sense. Done